kubeflow/trainer/rhai/traininghub.py

```
@@ -194,6 +793,23 @@ def get_trainer_cr_from_training_hub_trainer(
     else:
         # Secondary case: user provided function; embed their function and call with kwargs
         func_code, func_file = _render_user_func_code(trainer.func, trainer.func_args)
+
+        # Inject progress tracking code if enabled (for custom functions)
+        # Try to extract ckpt_output_dir from the function code or use default
+        if trainer.enable_progression_tracking and trainer.algorithm:
+            # For custom functions, use a default metrics directory
+            # User should ensure their function writes to this directory
+            ckpt_dir = "/tmp/training_metrics"
```

Ruff (0.14.8)

Probable insecure usage of temporary file or directory: "/tmp/training_metrics"

(S108)